### PR TITLE
[3.10] gh-97786: Fix compiler warnings in pytime.c (GH-101826)

### PR DESCRIPTION
**Pull request by mdickinson**
Fixes compiler warnings in pytime.c..
(cherry picked from commit b1b375e2670a58fc37cb4c2629ed73b045159918)

* Issue: gh-97786

### CODE REVIEW

File: Python/pytime.c [hunk 1]
```diff
@@ -722,7 +757,7 @@ py_get_system_clock(_PyTime_t *tp, _Py_clock_info_t *info, int raise)
         info->monotonic = 0;
         info->adjustable = 1;
         if (clock_getres(CLOCK_REALTIME, &res) == 0) {
-            info->resolution = res.tv_sec + res.tv_nsec * 1e-9;
+            info->resolution = (double)res.tv_sec + (double)res.tv_nsec * 1e-9;
```
Comment (abalkin) [inline Python/pytime.c:762]: It is not obvious why implicit conversion cannot be relied on here. Can you add a short comment explaining the need for explicit casts?

**Comment by gpshead** [comment]
!buildbot wasm

**Review by abalkin — APPROVED** [review]
Since this has already been merged in main (gh-97786), there isn't real need for a review.

In any case, LGTM.  I just added a suggestion that should not hold this from being merged.

File: Python/pytime.c (hunk 1 above)
Comment (mdickinson) [reply Python/pytime.c:762]: Yep, will do. For the record, it's purely to silence compiler warnings - nothing more than that. No numerical rationale.

Comment (mdickinson) [reply Python/pytime.c:762]: Done in 7e97e9118645a00dce198959bd93c752eacf0279

**Pull request merged**